fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/mixin/attachment/AttachmentTargetsMixin.java

```diff
@@ -47,6 +49,15 @@ public <T> T getAttached(AttachmentType<T> type) {
 	@Override
 	@Nullable
 	public <T> T setAttached(AttachmentType<T> type, @Nullable T value) {
+		// Extremely inelegant, but the only alternative is separating out these two mixins and duplicating code
+		Object thisObject = this;
+
+		if (thisObject instanceof BlockEntity) {
+			((BlockEntity) thisObject).markDirty();
+		} else if (thisObject instanceof WorldChunk) {
+			((Chunk) thisObject).setNeedsSaving(true);
+		}
```

Comment:
Users also need to do this manually for mutable data. This should be in the javadoc! Personally I would include target-specific documentation in `AttachmentTarget`. See for example what I did for NeoForge in `AttachmentType` where I list the specifics of every single attachment target.